node/src/components/consensus/highway_core/highway_testing.rs

```diff
@@ -353,8 +366,11 @@ where
         Ok(())
     }
 
-    fn next_consensus_value(&mut self) -> ConsensusValue {
-        self.consensus_values.pop_front().unwrap_or_default()
+    fn next_consensus_value(&mut self, height: u64) -> ConsensusValue {
+        self.consensus_values
+            .get(height as usize)
+            .cloned()
+            .unwrap_or_default()
```

Comment:
This is not _next_ consensus value anymore. Also, does it remove the value from the `consensus_value` queue? If not, does it mean we might propose the same value more than once?

Reply:
But that was the problem: That we removed it from the queue and proposed it only once, even if the block got orphaned. The existing tests only worked because they never orphan a block.

This way, it is in fact the _next_ consensus value: On height H it will always return entry number H, which is the one that comes after its parent.

Reply:
Ah, I see! The alternative implementation would detect orphaned consensus values and put it back in the queue.